### ISSUE TITLE
setupMocha `keepUnusedRequests` Config Option Isn't Saving All Requests to har File

### ISSUE THREAD

**Issue by Justintime50** [Netflix/pollyjs#455]
I'd expect when `keepUnusedRequests` is set to true that all requests would be recorded to the `har` file, even the unused ones. Printed to console is the following config which you'll notice doesn't include the `keepUnusedRequests` key. When using the `setupMocha` configuration in conjunction with `keepUnusedRequests`, I'm met with the following error:

### Error

```
    at Object.assert (/Users/jhammond/git/easypost/easypost-node/node_modules/@pollyjs/utils/src/utils/assert.js:5:11)
    at HttpAdapter.assert (/Users/jhammond/git/easypost/easypost-node/node_modules/@pollyjs/adapter/src/index.js:277:5)
    at HttpAdapter.<anonymous> (/Users/jhammond/git/easypost/easypost-node/node_modules/@pollyjs/adapter/src/index.js:144:10)
    at HttpAdapter.handleRequest (/Users/jhammond/git/easypost/easypost-node/node_modules/@pollyjs/adapter/src/index.js:92:13) {
  response: undefined
PollyError: [Polly] [adapter:node-http] Unhandled request:
```

This is the config that gets printed along with the error, notably not getting the `keepUnusedRequests` populated:

```json
"config": {
    "mode": "stopped",
    "adapters": [
      "node-http"
    ],
    "logLevel": 3,
    "flushRequestsOnStop": false,
    "recordIfMissing": true,
    "recordFailedRequests": false,
    "expiresIn": null,
    "expiryStrategy": "warn",
    "matchRequestsBy": {
      "method": true,
      "headers": false,
      "body": true,
      "order": true,
      "url": {
        "protocol": true,
        "username": true,
        "password": true,
        "hostname": true,
        "port": true,
        "pathname": true,
        "query": true,
        "hash": false
      }
    }
  }
```

### Config

```javascript
import { Polly, setupMocha as setupPolly } from '@pollyjs/core';

Polly.register(FSPersister);
Polly.register(NodeHttpAdapter);

function startPolly() {
  setupPolly({
    adapters: ['node-http'],
    persister: 'fs',
    persisterOptions: {
      keepUnusedRequests: true, // TODO: This isn't working
      fs: {
        recordingsDir: resolve(__dirname, './cassettes'),
      },
    },
    matchRequestsBy: {
      headers: false,
    },
  });
}
```

### Tests

My test file looks like the following (simplified) example:

```javascript
describe('Batch Resource', function () {
  setupPolly.startPolly();

  before(function () {
    this.easypost = new EasyPost(process.env.EASYPOST_TEST_API_KEY);
  });

  beforeEach(function () {
    const { server } = this.polly;
    setupPolly.stripCreds(server);
  });

  it('buys a batch', async function () {
    // NOTE: this gets recorded to the har file
    const batch = await new this.easypost.Batch({
      shipments: [Fixture.oneCallBuyShipment()],
    }).save();

    // NOTE: this should get recorded but doesn't
    batch.buy();

    expect(batch).to.be.an.instanceOf(this.easypost.Batch);
    expect(batch.num_shipments).to.equal(1);
  });
});
```

### Dependencies

I'll note I also tried this on `v5` of Pollyjs with the same results so it's probably not due to versioning but a configuration implementation?

```json
{
    "@pollyjs/adapter-node-http": "^6.0.4",
    "@pollyjs/core": "^6.0.4",
    "@pollyjs/persister-fs": "^6.0.4",
}
```

## Environment

```
Node.js v16.14.0
darwin 20.6.0
npm 8.3.1
```

**Comment by Justintime50** [comment]
I've done some additional testing and found that the `mode` for the second request in my test example above is getting set to `stopped` while the first request in the same test is set to `replay` which is what leads to the `Unhandled request` error as no other branch conditions were met.

Will continue testing but am unsure as to why it's getting stopped when there are no tests for the second call.

**Comment by Justintime50** [comment]
Let's pretend this never happened. Simply adding `await` to the requests ensured they were completed before Pollyjs closed the recording. It's also of note that this corrected things regardless of what `keepUnusedRequests` was set to meaning there's also no config change needed for untested requests. I assumed that this acted like Ruby's VCR where it complains about unused requests that don't have assertions against it.

Long story short, this was a great debugging adventure, thank you for the incredible tool, and there is no issue here 😉
